sentry_irc/plugin.py

```diff
@@ -114,43 +127,39 @@ def send_payload(self, project, message):
-        ircsock.send("USER %s %s %s :Sentry IRC bot\n" % ((nick,) * 3))
-        ircsock.send("NICK %s\n" % nick)
-        while (time.time() - start) < self.timeout:
-            ircmsg = ircsock.recv(2048).strip('\n\r')
-            real_nick = CONN_RE.search(ircmsg)
-            if real_nick is not None:
-                nick = real_nick.group(1)
-            pong = PING_RE.findall(ircmsg)
-            if pong:
-                ircsock.send("PONG %s\n" % pong)
-            if re.findall(' 433 \* %s' % nick, ircmsg):
-                nick += '%s' % randrange(1000, 2000)
-                ircsock.send("NICK %s\n" % nick)
+        ircsock = safe_socket_connect((server, int(port)), ssl=ssl_c)
+        try:
+            if password:
+                ircsock.send("PASS %s\n" % password)
+            ircsock.send("USER %s %s %s :Sentry IRC bot\n" % ((nick,) * 3))
+            ircsock.send("NICK %s\n" % nick)
+            while (time.time() - start) < self.timeout:
+                ircmsg = ircsock.recv(2048).strip('\n\r')
+                real_nick = CONN_RE.search(ircmsg)
+                if real_nick is not None:
+                    nick = real_nick.group(1)
+                pong = PING_RE.findall(ircmsg)
+                if pong:
+                    ircsock.send("PONG %s\n" % pong)
+                if re.findall(' 433 \* %s' % nick, ircmsg):
+                    nick += '%s' % randrange(1000, 2000)
+                    ircsock.send("NICK %s\n" % nick)
+                    ircmsg = ircsock.recv(2048)
+                if re.findall(' 00[1-4] %s' % nick, ircmsg):
+                    for room in rooms:
+                        if not without_join:
+                            ircsock.send("JOIN %s\n" % room)
+                        ircsock.send("PRIVMSG %s :%s\n" % (room, message))
+                        if not without_join:
+                            ircsock.send("PART %s\n" % room)
+                    for user in users:
+                        ircsock.send("PRIVMSG %s :%s\n" % (user, message))
+                    break
+
+            ircsock.send("QUIT\n")
+
+            # try to flush pending buffer
+            while (time.time() - start) < self.timeout and ircmsg:
                 ircmsg = ircsock.recv(2048)
-            if re.findall(' 00[1-4] %s' % nick, ircmsg):
-                for room in rooms:
-                    if not without_join:
-                        ircsock.send("JOIN %s\n" % room)
-                    ircsock.send("PRIVMSG %s :%s\n" % (room, message))
-                    if not without_join:
-                        ircsock.send("PART %s\n" % room)
-                for user in users:
-                    ircsock.send("PRIVMSG %s :%s\n" % (user, message))
-                break
-
-        ircsock.send("QUIT\n")
-
-        # try to flush pending buffer
-        while (time.time() - start) < self.timeout and ircmsg:
-            ircmsg = ircsock.recv(2048)
-        irc.close()
+        finally:
```

Comment:
:+1: for IRC network errors not blowing up. Would logging what failed be useful for the end user?

Reply:
Yes, but far outside the scope of this PR.